### PR TITLE
Editing options in Changeset dialog

### PR DESCRIPTION
**Pull request by GhostFoxSledgehammer**
New mapmode -Edit mode
Editing options disabled in select mode,
Support for changeset of type `deleteion`
New editing options in Changeset dialog
Empty shortcuts for undo/redo buttons

### CODE REVIEW

**Comment by GhostFoxSledgehammer** [comment]
@taylorsmock @tsmock please review

File: src/main/java/org/openstreetmap/josm/plugins/mapillary/mode/EditMode.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    } else if (closestTemp == null && highlightedImage != null) {
+      MapillaryLayer.getInstance().getData().setHighlightedImage(null);
+      MapillaryMainDialog.getInstance().setImage(MapillaryLayer.getInstance().getData().getSelectedImage());
+      MapillaryMainDialog.getInstance().updateImage();
+    }
+    MapillaryLayer.invalidateInstance();
+  }
+
+  @Override
+  public void paint(Graphics2D g, MapView mv, Bounds box) {
```
Comment (taylorsmock) [inline src/main/java/org/openstreetmap/josm/plugins/mapillary/mode/EditMode.java:212]: You should probably explain why this is empty. For future people. :)

**Comment by taylorsmock** [comment]
Also, if there are imported images, you _may_ want to at least point people to the correct location to do the upload.

**Comment by GhostFoxSledgehammer** [comment]
> 
> 
> Also, if there are imported images, you _may_ want to at least point people to the correct location to do the upload.

Like a walk mode but only for Images that are being uploaded, then mark each image as ready for upload and then final upload?

File: src/main/java/org/openstreetmap/josm/plugins/mapillary/gui/changeset/SubmitLocationChangesetAction.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    MainApplication.worker.submit(() -> this.submitChangeset());
+  }
+
+  public void submitChangeset() {
+    changesetDialog.setUploadPending(true);
+    String token = MapillaryProperties.ACCESS_TOKEN.get();
+    if (token != null && !token.trim().isEmpty()) {
+      PluginState.setSubmittingChangeset(true);
+      MapillaryUtils.updateHelpText();
+      HttpClientBuilder builder = HttpClientBuilder.create();
```
[the remark this answers is not in the document]
Comment (GhostFoxSledgehammer) [reply src/main/java/org/openstreetmap/josm/plugins/mapillary/gui/changeset/SubmitLocationChangesetAction.java:75]: Ok, how do I add apache logging to the project?

**Comment by taylorsmock** [comment]
> > Also, if there are imported images, you _may_ want to at least point people to the correct location to do the upload.
> 
> Like a walk mode but only for Images that are being uploaded, then mark each image as ready for upload and then final upload?

Like say "Imported Images Detected, to upload go to File -> Upload Mapillary Images".

File: src/main/java/org/openstreetmap/josm/plugins/mapillary/gui/changeset/SubmitLocationChangesetAction.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tsmock) [reply src/main/java/org/openstreetmap/josm/plugins/mapillary/gui/changeset/SubmitLocationChangesetAction.java:75]: We would want to add `ivy` as a dependency resolution tool. See MapWithAI for an example. (`build.gradle`, `ivy.xml`, and `build.xml` all need to be added or modified).

**Pull request merged**